packages/block-library/src/post-featured-image/index.php

```diff
@@ -19,6 +19,12 @@ function render_block_core_post_featured_image( $attributes, $content, $block )
 	}
 	$post_ID = $block->context['postId'];
 
+	// Check is needed for backward compatibility with third-party plugins
+	// that might rely on the `in_the_loop` check; calling `the_post` sets it to true.
+	if ( ! in_the_loop() && have_posts() ) {
+		the_post();
```

Comment:
same question as the core API, why not just set the in_the_loop global directly (the_post might have unintended consequences?)

Reply:
@adamsilverstein I think setting only the in_the_loop global would have other unintended consequences.

To be fair, this code is not new, I copied it as is from the `post-content` block where this is already used in production since 1.5 years (see https://github.com/WordPress/wordpress-develop/commit/585f62ad8c929af0100467ce2d51fed749a02b4f#diff-7adcebab0e7d7e6e897a5854fd02fc2421ea6acfc1ccc21c651fde4ef5c0a3caR47, this is when it was introduced in core). Given that nobody has raised an issue about it since, I felt it was reasonable enough to reuse the same approach here.